internal/commands/pull/inputs_test.go

```diff
@@ -116,6 +132,22 @@ func TestPullInputsResolveRemoteApp(t *testing.T) {
 		assert.Equal(t, realm.AppFilter{GroupID: "some-project", App: "some-app"}, appFilter)
 	})
 
+	t.Run("should not resolve group id if app meta exists", func(t *testing.T) {
+		i := inputs{appMeta: local.AppMeta{GroupID: "some-group", AppID: "some-app", ConfigVersion: realm.DefaultAppConfigVersion}}
+		var realmClient mock.RealmClient
+
+		var findAppCalls int
+		realmClient.FindAppFn = func(groupID, appID string) (realm.App, error) {
+			findAppCalls++
```

Comment:
[q] why use an int here instead of a bool?

Reply:
it's actually testing that we only call it once rather than twice, so consider this like a "super bool" :)